<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>713</width>
    <height>551</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Plantscan3D</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="4" colspan="6">
     <widget class="GLMTGEditor" name="mtgeditor" native="true"/>
    </item>
    <item row="0" column="6">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Clip</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QCheckBox" name="visibilityEnabled">
      <property name="text">
       <string/>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="6">
     <widget class="QSlider" name="backVisibilitySlider">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="invertedControls">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="6">
     <widget class="QSlider" name="frontVisibilitySlider">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>PointSize</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QSlider" name="pointSizeSlider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>PointFilter</string>
      </property>
     </widget>
    </item>
    <item row="4" column="3">
     <widget class="QSlider" name="pointFilterSlider">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>4</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>NodeSize</string>
      </property>
     </widget>
    </item>
    <item row="4" column="5" colspan="2">
     <widget class="QSlider" name="nodeSizeSlider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>713</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport_MTG">
     <property name="title">
      <string>Export MTG</string>
     </property>
     <addaction name="actionExportNodeList"/>
    </widget>
    <widget class="QMenu" name="menuExport_View">
     <property name="title">
      <string>Export View</string>
     </property>
     <addaction name="actionExportGeom"/>
     <addaction name="actionSaveSnapshot"/>
    </widget>
    <widget class="QMenu" name="menuLoad">
     <property name="title">
      <string>Load</string>
     </property>
     <addaction name="actionPuu1"/>
     <addaction name="actionPuu3"/>
     <addaction name="actionCherry"/>
     <addaction name="actionArabido"/>
     <addaction name="actionAppleTree"/>
    </widget>
    <widget class="QMenu" name="menuRecents">
     <property name="title">
      <string>Recents</string>
     </property>
    </widget>
    <addaction name="actionOpen_Project"/>
    <addaction name="actionSave_Project"/>
    <addaction name="separator"/>
    <addaction name="actionOpenMTG"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveMTG"/>
    <addaction name="separator"/>
    <addaction name="actionImportPoints"/>
    <addaction name="actionExportPoints"/>
    <addaction name="separator"/>
    <addaction name="actionImportMTG"/>
    <addaction name="menuExport_MTG"/>
    <addaction name="separator"/>
    <addaction name="menuExport_View"/>
    <addaction name="separator"/>
    <addaction name="menuRecents"/>
    <addaction name="menuLoad"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuTheme">
     <property name="title">
      <string>Theme</string>
     </property>
     <addaction name="actionWhiteTheme"/>
     <addaction name="actionBlackTheme"/>
    </widget>
    <addaction name="actionViewPoints"/>
    <addaction name="actionViewPointAttributes"/>
    <addaction name="separator"/>
    <addaction name="actionViewMTG"/>
    <addaction name="actionViewControlPoints"/>
    <addaction name="actionViewRadius"/>
    <addaction name="separator"/>
    <addaction name="actionView3DModel"/>
    <addaction name="separator"/>
    <addaction name="actionRefreshView"/>
    <addaction name="separator"/>
    <addaction name="actionAdjustView"/>
    <addaction name="menuTheme"/>
   </widget>
   <widget class="QMenu" name="menuReconstruction">
    <property name="title">
     <string>Reconstruction</string>
    </property>
    <widget class="QMenu" name="menuRadius">
     <property name="title">
      <string>Radius</string>
     </property>
     <addaction name="actionComputeMaxRadius"/>
     <addaction name="actionComputeMeanRadius"/>
     <addaction name="separator"/>
     <addaction name="actionSmoothRadius"/>
     <addaction name="actionThresholdRadius"/>
     <addaction name="separator"/>
     <addaction name="actionPipeModel"/>
     <addaction name="actionPipeModelAverageDistance"/>
    </widget>
    <widget class="QMenu" name="menuAdd_Root">
     <property name="title">
      <string>Add Root</string>
     </property>
     <addaction name="actionRootBottom"/>
     <addaction name="actionRootTop"/>
    </widget>
    <widget class="QMenu" name="menuSkeletization">
     <property name="title">
      <string>Skeletization</string>
     </property>
     <addaction name="actionXuReconstruction"/>
     <addaction name="actionGraphColonization"/>
     <addaction name="actionLivnyReconstruction"/>
     <addaction name="actionScaReconstruction"/>
     <addaction name="actionAdaptiveScaReconstruction"/>
    </widget>
    <widget class="QMenu" name="menuMultiScale">
     <property name="title">
      <string>Multi Scale</string>
     </property>
     <addaction name="actionEditScale"/>
     <addaction name="actionCommitScale"/>
    </widget>
    <widget class="QMenu" name="menuSkeleton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Skeleton</string>
     </property>
     <addaction name="actionSmoothPosition"/>
     <addaction name="actionCheckMTG"/>
    </widget>
    <addaction name="menuAdd_Root"/>
    <addaction name="separator"/>
    <addaction name="menuSkeletization"/>
    <addaction name="separator"/>
    <addaction name="menuSkeleton"/>
    <addaction name="separator"/>
    <addaction name="menuRadius"/>
    <addaction name="separator"/>
    <addaction name="action3DRepresentation"/>
    <addaction name="separator"/>
    <addaction name="menuMultiScale"/>
    <addaction name="separator"/>
    <addaction name="actionTagProperty"/>
    <addaction name="separator"/>
    <addaction name="actionAngleEstimation"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionRevolveAroundScene"/>
    <addaction name="actionShowAll"/>
   </widget>
   <widget class="QMenu" name="menuPoints">
    <property name="title">
     <string>Points</string>
    </property>
    <widget class="QMenu" name="menuContraction">
     <property name="title">
      <string>Contraction</string>
     </property>
     <addaction name="actionEuclidianContraction"/>
     <addaction name="actionLaplacianContraction"/>
     <addaction name="actionRiemannianContraction"/>
     <addaction name="actionAdaptiveRadialContraction"/>
     <addaction name="actionPathBasedContraction"/>
     <addaction name="actionROSAContraction"/>
    </widget>
    <widget class="QMenu" name="menuAlign">
     <property name="title">
      <string>Align</string>
     </property>
     <addaction name="actionAlignGlobal"/>
     <addaction name="separator"/>
     <addaction name="actionScaleAndCenter"/>
     <addaction name="separator"/>
     <addaction name="actionAlignOptimizeAll"/>
     <addaction name="actionAlignOptimizeOrientation"/>
     <addaction name="actionAlignOptimizePosition"/>
    </widget>
    <widget class="QMenu" name="menuDensity">
     <property name="title">
      <string>Density</string>
     </property>
     <addaction name="actionKDensity"/>
     <addaction name="actionRDensity"/>
     <addaction name="actionRDensity_MT"/>
     <addaction name="separator"/>
     <addaction name="actionDensityHistogram"/>
    </widget>
    <widget class="QMenu" name="menuFilter">
     <property name="title">
      <string>Filter</string>
     </property>
     <addaction name="actionFilterMinDensity"/>
     <addaction name="actionFilterMaxDensity"/>
    </widget>
    <widget class="QMenu" name="menuDirections">
     <property name="title">
      <string>Directions</string>
     </property>
     <addaction name="actionPointDirections"/>
     <addaction name="actionAverageDirections"/>
    </widget>
    <addaction name="actionReorient"/>
    <addaction name="actionSortZ"/>
    <addaction name="separator"/>
    <addaction name="actionSubSampling"/>
    <addaction name="separator"/>
    <addaction name="menuDensity"/>
    <addaction name="separator"/>
    <addaction name="menuDirections"/>
    <addaction name="separator"/>
    <addaction name="actionPointNormals"/>
    <addaction name="separator"/>
    <addaction name="menuFilter"/>
    <addaction name="separator"/>
    <addaction name="menuContraction"/>
    <addaction name="separator"/>
    <addaction name="menuAlign"/>
    <addaction name="separator"/>
    <addaction name="actionPointClusters"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuPoints"/>
   <addaction name="menuReconstruction"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNodesInfo">
   <property name="text">
    <string>Import Debug Info</string>
   </property>
  </action>
  <action name="actionImportPoints">
   <property name="text">
    <string>Import Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionOpenMTG">
   <property name="text">
    <string>Open MTG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveMTG">
   <property name="text">
    <string>Save MTG as</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionViewMTG">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>MTG</string>
   </property>
  </action>
  <action name="actionViewPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Points</string>
   </property>
  </action>
  <action name="actionAdjustView">
   <property name="text">
    <string>Adjust View</string>
   </property>
  </action>
  <action name="actionViewControlPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Control Points</string>
   </property>
  </action>
  <action name="actionRefreshView">
   <property name="text">
    <string>Refresh View</string>
   </property>
  </action>
  <action name="actionContract">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Preuksakarn et al.</string>
   </property>
  </action>
  <action name="actionAdaptiveScaReconstruction">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Adaptive Space Colonization</string>
   </property>
  </action>
  <action name="actionExportGeom">
   <property name="text">
    <string>As PlantGL file</string>
   </property>
  </action>
  <action name="actionViewContractedPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Contracted Points</string>
   </property>
  </action>
  <action name="actionImportContractedPoints">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Import Contracted Points</string>
   </property>
  </action>
  <action name="actionExportContractedPoints">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Export Contracted Points</string>
   </property>
  </action>
  <action name="actionView3DModel">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>3D Model</string>
   </property>
  </action>
  <action name="actionComputeMaxRadius">
   <property name="text">
    <string>Estimate As Max Point Dist.</string>
   </property>
  </action>
  <action name="actionSmoothRadius">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Smooth</string>
   </property>
  </action>
  <action name="actionThresholdRadius">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Filtering</string>
   </property>
  </action>
  <action name="actionCheckMTG">
   <property name="text">
    <string>Check Validity</string>
   </property>
  </action>
  <action name="actionExportNodeList">
   <property name="text">
    <string>As Node List</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionPuu1">
   <property name="text">
    <string>puu1</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="actionPuu3">
   <property name="text">
    <string>puu3</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+3</string>
   </property>
  </action>
  <action name="actionCherry">
   <property name="text">
    <string>cherry</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+9</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save MTG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveSnapshot">
   <property name="text">
    <string>As snapshot</string>
   </property>
  </action>
  <action name="actionRevolveAroundScene">
   <property name="text">
    <string>Revolve Around Scene</string>
   </property>
  </action>
  <action name="actionShowAll">
   <property name="text">
    <string>Show All</string>
   </property>
  </action>
  <action name="actionReorient">
   <property name="text">
    <string>Swap Y and Z</string>
   </property>
  </action>
  <action name="actionXuReconstruction">
   <property name="text">
    <string>Xu et al.</string>
   </property>
  </action>
  <action name="actionRootBottom">
   <property name="text">
    <string>Bottom</string>
   </property>
  </action>
  <action name="actionRootTop">
   <property name="text">
    <string>Top</string>
   </property>
  </action>
  <action name="actionSubSampling">
   <property name="text">
    <string>Sub-Sampling</string>
   </property>
  </action>
  <action name="actionTest">
   <property name="text">
    <string>test</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionEuclidianContraction">
   <property name="text">
    <string>Euclidian</string>
   </property>
  </action>
  <action name="actionRiemannianContraction">
   <property name="text">
    <string>Riemannian</string>
   </property>
  </action>
  <action name="actionAngleEstimation">
   <property name="text">
    <string>Angle Estimation</string>
   </property>
  </action>
  <action name="actionAs_nothing">
   <property name="text">
    <string>As nothing</string>
   </property>
  </action>
  <action name="actionAs_picture">
   <property name="text">
    <string>As picture</string>
   </property>
  </action>
  <action name="actionExportPoints">
   <property name="text">
    <string>Export Points</string>
   </property>
  </action>
  <action name="actionOpen_Project">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Open Project</string>
   </property>
  </action>
  <action name="actionSave_Project">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save Project</string>
   </property>
  </action>
  <action name="actionArabido">
   <property name="text">
    <string>arabido</string>
   </property>
  </action>
  <action name="actionEditScale">
   <property name="text">
    <string>Scale Edition</string>
   </property>
  </action>
  <action name="actionCommitScale">
   <property name="text">
    <string>Commit Scale</string>
   </property>
  </action>
  <action name="actionTagProperty">
   <property name="text">
    <string>Tag Property</string>
   </property>
  </action>
  <action name="actionAlignGlobal">
   <property name="text">
    <string>Globally</string>
   </property>
  </action>
  <action name="actionAppleTree">
   <property name="text">
    <string>apple tree</string>
   </property>
  </action>
  <action name="actionAlignOptimizeAll">
   <property name="text">
    <string>Optimize All</string>
   </property>
  </action>
  <action name="actionAlignOptimizeOrientation">
   <property name="text">
    <string>Optimize Orientation</string>
   </property>
  </action>
  <action name="actionAlignOptimizeSize">
   <property name="text">
    <string>Optimize Size</string>
   </property>
  </action>
  <action name="actionAlignOptimizePosition">
   <property name="text">
    <string>Optimize Position</string>
   </property>
  </action>
  <action name="actionViewRadius">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Radius</string>
   </property>
  </action>
  <action name="actionScale_and_Center">
   <property name="text">
    <string>Scale and Center</string>
   </property>
  </action>
  <action name="actionScaleAndCenter">
   <property name="text">
    <string>Scale and Center</string>
   </property>
  </action>
  <action name="action3DRepresentation">
   <property name="text">
    <string>3D Representation</string>
   </property>
  </action>
  <action name="actionComputeMeanRadius">
   <property name="text">
    <string>Estimate As Mean Point Dist</string>
   </property>
  </action>
  <action name="actionImportMTG">
   <property name="text">
    <string>Import Digit MTG</string>
   </property>
  </action>
  <action name="actionLaplacianContraction">
   <property name="text">
    <string>Laplacian</string>
   </property>
  </action>
  <action name="actionAdaptiveRadialContraction">
   <property name="text">
    <string>Adaptive Radial</string>
   </property>
  </action>
  <action name="actionFilterMinDensity">
   <property name="text">
    <string>Filter Min Density</string>
   </property>
  </action>
  <action name="actionFilterMaxDensity">
   <property name="text">
    <string>Filter Max Density</string>
   </property>
  </action>
  <action name="actionKDensity">
   <property name="text">
    <string>K-Density</string>
   </property>
  </action>
  <action name="actionRDensity">
   <property name="text">
    <string>R-Density</string>
   </property>
  </action>
  <action name="actionLivnyReconstruction">
   <property name="text">
    <string>Livny et al</string>
   </property>
  </action>
  <action name="actionRDensity_MT">
   <property name="text">
    <string>R-Density MT</string>
   </property>
  </action>
  <action name="actionPathBasedContraction">
   <property name="text">
    <string>Path Based (Livny et al)</string>
   </property>
  </action>
  <action name="actionPointNormals">
   <property name="text">
    <string>Normals</string>
   </property>
  </action>
  <action name="actionROSAContraction">
   <property name="text">
    <string>ROSA (Tagliasacchi et al.)</string>
   </property>
  </action>
  <action name="actionViewPointAttributes">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Point Attributes</string>
   </property>
  </action>
  <action name="actionSortZ">
   <property name="text">
    <string>Sort Z</string>
   </property>
  </action>
  <action name="actionSmoothPosition">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Smooth</string>
   </property>
  </action>
  <action name="actionPipeModelAverageDistance">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pipe Model from Av. Dist</string>
   </property>
  </action>
  <action name="actionPipeModel">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pipe Model</string>
   </property>
  </action>
  <action name="actionDensityHistogram">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Histogram</string>
   </property>
  </action>
  <action name="actionPointDirections">
   <property name="text">
    <string>Estimate</string>
   </property>
  </action>
  <action name="actionAverageDirections">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Average</string>
   </property>
  </action>
  <action name="actionClearRecents">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionGraphColonization">
   <property name="text">
    <string>Adaptive R. Graph Colonization</string>
   </property>
  </action>
  <action name="actionPointClusters">
   <property name="text">
    <string>Clusters</string>
   </property>
  </action>
  <action name="actionWhiteTheme">
   <property name="text">
    <string>White</string>
   </property>
  </action>
  <action name="actionBlackTheme">
   <property name="text">
    <string>Black</string>
   </property>
  </action>
  <action name="actionScaReconstruction">
   <property name="text">
    <string>Space Colonization</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLMTGEditor</class>
   <extends>QWidget</extends>
   <header>mtgeditorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>visibilityEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>frontVisibilitySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>visibilityEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>backVisibilitySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
